### ISSUE TITLE
Add link rel=canonical for requests routed with MapProxyDomain

### ISSUE THREAD

**Issue by ademarre** [apache/incubator-pagespeed-mod#1379]
The link rel=canonical header has been working for images and PDFs as of #1184. Apparently mod_pagespeed doesn't add the header for requests routed with MapProxyDomain, which is an important use case for me.

**Comment by jeffkaufman** [comment]
How would you want this to work? Say you have:

```
MapProxyDomain a.com b.com
```

When requesting `a.com/foo` would you like to see a `rel=canonical` pointing to `b.com/foo`?

If so, that would at least have to be opt-in, since many existing `MapProxyDomain` users probably view where they're proxying from as a private implementation detail that they don't want shared, and don't want search engines to care about.

**Comment by ademarre** [comment]
Yes, I would want it to work just like you said. And I agree it should probably be opt-in.

The `rel=canonical` URL should be the one that would be used if PageSpeed wasn't rewriting. This would mean preserving the `rel=canonical` link header if it already exists at the origin, and using the original URL otherwise.

I have a scenario where I'm running my website with PageSpeed at `www.example.com` and my other assets (JS, CSS, PDFs, images) are at `static.example.com`. I can't run PageSpeed at `static.example.com`, so I'm proxying them in order to enable rewriting.

```
MapProxyDomain www.example.com/proxy/static static.example.com
```

`https://www.example.com/proxy/static/images/xcompany-photo.jpg.pagespeed.ic.Ryj3GdrwtD.webp` is not a clean canonical URL, and it will change when I modify the image. I would consider `https://static.example.com/images/company-photo.jpg` to be canonical.
